src/main.rs

```diff
@@ -80,13 +82,33 @@ impl Valgrind {
         Ok(())
     }
     fn is_in_bounds(&self, addr: usize, len: usize) -> bool {
-        addr + len <= self.metadata.len()
+        addr + len <= self.metadata.len() && self.max_stack_size < addr
```

Comment:
As far as I can tell, I think everything should "just work" for accesses to the stack as well if we don't add this condition, right? In other words, given that we update `metadata` appropriately in grow/shrink stack functions below, we should be able to go through the same paths for both?

Reply:
I had added this condition with the intent to catch attempted mallocs/frees in memory dedicated to the stack, but am now also realizing that it would be blocking valid stack read/writes (if read/writes to stack are valid in this context that is?). Would it be appropriate to move the `&& self.max_stack_size < addr` condition to free/malloc instead?